scipy/special/_basic.pyi

```diff
@@ -0,0 +1,240 @@
+from __future__ import annotations
+
+from typing import (
+    Callable,
+    overload,
+    Literal,
+)
+
+import numpy as np
+
+from ._ufuncs import psi
+from numpy import sinc as sinc
+
+from numpy.typing import ArrayLike, NDArray
+
+_IntegerType = int | np.integer
+_FloatingType = float | np.floating
+_ComplexNumber = complex | _FloatingType | _IntegerType
+
+_PointsAndDerivatives = tuple[NDArray, NDArray]
+_ZerosOrder = tuple[NDArray, NDArray, NDArray, NDArray]
```

Comment:
The `NDArray` aliases are missing a type argument or type parameter. Type-checkers will otherwise infer it as `numpy.ndarray[tuple[int, ...], numpy.dtype[Unknown]]`, which e.g. pyright in strict mode reports as an error. 

So you could either pass a "fixed" scalar type, like `NDArray[np.float64]`, or parametrize them as e.g. 

```
_SCT = TypeVar("_SCT", bound=np.generic)

_PointsAndDerivatives: TypeAlias = tuple[npt.NDArray[_SCT]]
```

so that you can use the alias as e.g. `_PointsAndDerivatives[np.float64]` or `_PointsAndDerivatives[np.complex128]`

You could even use PEP 696 already through `typing_extensions` (always available in typing stubs, even if the `typing_extensions` isn't available at runtime), to add a default to `_SCT`:

```
from typing_extensions import TypeVar

_SCT = TypeVar("_SCT", bound=np.generic, default=np.float64)

...
```

This'd allow you to write `_PointsAndDerivatives`, which static type-checkers will interpret as `_PointsAndDerivatives[np.float64]`.